File: 02_Repeated_String.cpp
//
// Created by Jan Engelke on 08/12/2019.
// Taken from
// https://www.hackerrank.com/challenges/repeated-string
//

#include <bits/stdc++.h>

using namespace std;

// Complete the repeatedString function below.
long repeatedString(string s, long n) {

    long int_div_num = (n / s.size());

    long remainder = n % s.size();
    string substring = s.substr(0, remainder);

    long string_count = count(s.begin(), s.end(), 'a');
    long substring_count = count(substring.begin(), substring.end(), 'a');

    long count = string_count * int_div_num + substring_count;

    return count;

}

File: 02_Repeated_String.py
"""
https://www.hackerrank.com/challenges/repeated-string
"""
import math
import os
import random
import re
import sys

# Complete the repeatedString function below.

s = "abcac"
n = 10


def repeatedString(s, n):
    if len(s) == 1:
        if s == "a":
            return n
        else:
             return 0
    count = s.count("a") * (n // len(s)) + s[: n % len(s)].count("a")
    return count

print(repeatedString(s, n))
